scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/TensorFlowBucketMetadata.java

```diff
@@ -67,6 +67,14 @@ public K extractKey(Example value) {
       BytesList values = feature.getBytesList();
       Preconditions.checkState(values.getValueCount() == 1, "Number of feature in keyField != 1");
       return (K) values.getValue(0).toByteArray();
+    } else if (getKeyClass() == ByteString.class) {
+      BytesList values = feature.getBytesList();
+      Preconditions.checkState(values.getValueCount() == 1, "Number of feature in keyField != 1");
+      return (K) ByteString.copyFrom(values.getValue(0).toByteArray());
```

Comment:
any reason why we copy it instead of just `values.getValue(0)`?

Reply:
Good find. Shouldn't have to copy as `ByteString` is immutable.

Reply:
```suggestion
      return (K) values.getValue(0);
```